src/lexedata/importer/fromexcel.py

```diff
@@ -603,51 +610,57 @@ def properties_from_row(self, row: t.List[openpyxl.cell.Cell]) -> Row:
 
 
 def load_dataset(
-    metadata: Path, lexicon: str, db: Path, cognate_lexicon: t.Optional[str]
+    metadata: Path,
+    lexicon: t.Optional[str],
+    cognate_lexicon: t.Optional[str] = None,
 ):
+
     # logging.basicConfig(filename="warnings.log")
     dataset = pycldf.Dataset.from_metadata(metadata)
-    if db == "":
-        tmpdir = Path(mkdtemp("", "fromexcel"))
-        db = tmpdir / "db.sqlite"
-    lexicon_wb = openpyxl.load_workbook(lexicon).active
-    # check for dialect.cognate in metadata
-    dialect_cognate = False
+    # load dialect from metadata
     try:
         dialect = argparse.Namespace(
             **dataset.tablegroup.common_props["special:fromexcel"]
         )
-        dialect_cognate = True if dialect.cognates else False
     except KeyError:
         logger.warning(
             "Dialect not found or dialect was missing a key, "
             "falling back to default parser"
         )
         dialect = None
-    # load lexical data set
-    if dialect:
-        EP = excel_parser_from_dialect(dataset, dialect, cognate=False)
-    else:
-        EP = ExcelParser
-    # The Intermediate Storage, in a in-memory DB (unless specified otherwise)
-    EP = EP(dataset, db_fname=db)
-    EP.db.empty_cache()
-    EP.parse_cells(lexicon_wb)
+
+    if lexicon:
```

Comment:
I think the function load_dataset as it is right now in master is better than this one. It seems like you could run this one without any effect.